cfpb/sbl-frontend
# feat(e2e): sets up playwright, 1st user flow test

Adds playwright to the test suite including deps, config, test fixtures, and sample tests. This PR gets improved upon by other pending PRs.

Closes: #709 

## Changes

(in order of [files changed](https://github.com/cfpb/sbl-frontend/pull/776/files))

- add playwright files to `.gitignore`
- add yarn specific config to `.npmrc` for playright/vscode integration
- add yarn pnp specific config to `settings.json`
- add very basic test instructions to the `readme.md`
- provide a sample test to `example.spec.demo.ts` to show different methods available to use during testing
- add `textFixture.ts` that spins up new test accounts and provides convenience functions to navigate through the app
- add `completeFilngFlowWithOnlyWarnings.spec.ts` to test through sign and submit
- add `sampleTestCfpbLogoGoesToConsumerFinance.spec.ts` as a basic test to make sure site loads
- add `point-of-contact-data-1.json` for basic point of contact info
- add  `sbl-validations-all-pass-small.csv` as a test file
- creates utility functions to drive the fixtures
  - `createDomainAssociation.ts`
  - `createInstitution.ts`
  - `createKeycloakUser.ts`
  - `getKeycloakToken.ts`
- add `playwright.config.ts` with just the basic configuration and only using chrome for now
- change `Loading.tsx` to be a paragraph instead of a heading since we don't need to have a heading for unstructured loading content (and it also helps with all the tests)

## Notes
- lots of PRs depend on this one, so some things have been improved in later PRs
- there are typescript errors that will need to be fixed, but there's a [yarn pnp bug](https://github.com/microsoft/playwright/issues/21058) that keeps types from being easily imported. We'll fix it up in a future PR, and when we switch to node_modules, we might get the types imported easily too.
- eslint gets fixed [in a later PR](https://github.com/cfpb/sbl-frontend/pull/821), so we'll clean it up after that PR gets merged
- Right now the fixture doesn't clean up the generated users/institutions, but we're just running these tests locally for now, and we'll be having conversations about adding APIs for that on Monday
- this is just the start of more playwright awesomeness! 😎

## How to test this PR

1. Run `yarn playwright test --ui`, see if the tests pass
2. Does the site still operate normally otherwise?


## Screenshots
<img width="1392" alt="Screenshot 2024-07-25 at 11 04 34 AM" src="https://github.com/user-attachments/assets/eb791594-994a-4dd8-a450-34f0e7724cfe">
